sdk/core/azure-core/src/main/java/com/azure/core/implementation/ResponseConstructorsCache.java

```diff
@@ -96,7 +97,8 @@
                         responseStatusCode,
                         responseHeaders));
                 } catch (IllegalAccessException | InvocationTargetException | InstantiationException e) {
-                    throw logger.logExceptionAsError(Exceptions.propagate(e));
+                    throw logger.logExceptionAsError(new RuntimeException("Failed to deserialize 3-parameter "
+                        + "response. Error Details: " + e.getMessage(), e));
```

Comment:
Are we logging twice here when these fail? Once here and again for failing to deserialize response.

Additionally I'm not sure we should log these so specifically, it seems failing to decide response is sufficient.

Reply:
Good catch. The error message show up in stack trace twice. Removed the one in each block. 

Having the parameter info is to help developer quickly determine which response constructor failed.